utils/src/main/scala/za/co/absa/enceladus/utils/testUtils/SparkTestBase.scala

```diff
@@ -17,21 +17,117 @@ package za.co.absa.enceladus.utils.testUtils
 
 import org.apache.log4j.{Level, Logger}
 import org.apache.spark.sql.SparkSession
+import org.apache.spark.SparkConf
+import com.typesafe.config.ConfigFactory
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{Path, FileSystem}
+import scala.collection.JavaConversions._
+import java.io.File
 import za.co.absa.enceladus.utils.time.TimeZoneNormalizer
+import com.typesafe.config.Config
 
-trait SparkTestBase {
-  System.setProperty("user.timezone", "UTC");
+trait SparkTestBase { self =>
+  System.setProperty("user.timezone", "UTC")
+
+  val config: Config = ConfigFactory.load()
+  val sparkMaster: String = config.getString("enceladus.utils.testUtils.sparkTestBaseMaster")
+
+  val sparkBuilder: SparkSession.Builder = SparkSession.builder()
+    .master(sparkMaster)
+    .appName(s"Enceladus test - ${self.getClass.getName}")
+    .config("spark.ui.enabled", "false")
+
+  implicit val spark: SparkSession = if (sparkMaster == "yarn") {
+    val confDir = config.getString("enceladus.utils.testUtils.hadoop.conf.dir")
+    val distJarsDir = config.getString("enceladus.utils.testUtils.spark.distJars.dir")
+    val sparkHomeDir = config.getString("enceladus.utils.testUtils.spark.home.dir")
+
+    val hadoopConfigs = SparkTestBase.getHadoopConfigurationForSpark(confDir)
+    val sparkConfigs = SparkTestBase.loadSparkDefaults(sparkHomeDir)
+    val allConfigs = hadoopConfigs ++ sparkConfigs
+
+    //get a list of all dist jars
+    val distJars = FileSystem.get(SparkTestBase.getHadoopConfiguration(confDir)).listStatus(new Path(distJarsDir)).map(_.getPath)
+    val localJars = SparkTestBase.getDepsFromClassPath("absa")
+    val currentJars = SparkTestBase.getCurrentProjectJars
+    val deps = (distJars ++ localJars ++currentJars).mkString(",")
+
+    sparkBuilder.config(new SparkConf().setAll(allConfigs))
+      .config("spark.yarn.jars", deps)
+      .config("spark.deploy.mode", "client")
+      .getOrCreate()
+
+  } else {
+    sparkBuilder
+      .config("spark.driver.bindAddress", "127.0.0.1")
+      .config("spark.driver.host", "127.0.0.1")
+      .getOrCreate()
+  }
 
-  implicit val spark = SparkSession.builder().master("local[*]").appName("test")
-  .config("spark.sql.codegen.wholeStage", false)
-  .config("spark.ui.enabled", "false")
-  .config("spark.driver.bindAddress","127.0.0.1")
-  .config("spark.driver.host", "127.0.0.1")
-  .getOrCreate()
   TimeZoneNormalizer.normalizeAll(Seq(spark))
 
   // Do not display INFO entries for tests
   Logger.getLogger("org").setLevel(Level.WARN)
   Logger.getLogger("akka").setLevel(Level.WARN)
 
 }
+
+object SparkTestBase {
+  /**
+   * Gets a Hadoop configuration object from the specified hadoopConfDir parameter
+   * 
+   * @param hadoopConfDir string representation of HADOOP_CONF_DIR
+   */
+  def getHadoopConfiguration(hadoopConfDir: String): Configuration = {
```

Comment:
Tiny: I believe the functions could be made `private` for clarity and to speed up compilation.